server/rest/src/main/java/org/infinispan/rest/server/operations/CacheOperations.java

```diff
@@ -0,0 +1,303 @@
+package org.infinispan.rest.server.operations;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.CacheSet;
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.hash.MurmurHash3;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.container.entries.InternalCacheEntry;
+import org.infinispan.metadata.Metadata;
+import org.infinispan.rest.MimeMetadata;
+import org.infinispan.rest.MimeMetadataBuilder;
+import org.infinispan.rest.RestCacheManager;
+import org.infinispan.rest.configuration.RestServerConfiguration;
+import org.infinispan.rest.server.CacheControl;
+import org.infinispan.rest.server.InfinispanRequest;
+import org.infinispan.rest.server.InfinispanResponse;
+import org.infinispan.rest.server.operations.exceptions.NoCacheFoundException;
+import org.infinispan.rest.server.operations.exceptions.NoDataFoundException;
+import org.infinispan.rest.server.operations.exceptions.NoKeyException;
+import org.infinispan.rest.server.operations.exceptions.NoMediaTypeException;
+import org.infinispan.rest.server.operations.exceptions.RestResponseException;
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+public class CacheOperations {
+
+   private static final MurmurHash3 hashFunc = MurmurHash3.getInstance();
+
+   private final RestCacheManager restCacheManager;
+   private final RestServerConfiguration restServerConfiguration;
+
+   public CacheOperations(RestServerConfiguration configuration, RestCacheManager cacheManager) {
+      this.restServerConfiguration = configuration;
+      this.restCacheManager = cacheManager;
+   }
+
+   public InfinispanResponse getCacheValues(InfinispanRequest request) throws RestResponseException {
+      try {
+         String cacheName = request.getCacheName();
+         AdvancedCache<String, byte[]> cache = restCacheManager.getCache(cacheName);
+         CacheSet<String> keys = cache.keySet();
+         MediaType mediaType = request.getMediaType()
+               .map(m -> MediaType.fromMediaTypeAsString(m))
+               .orElseThrow(NoMediaTypeException::new);
+
+         InfinispanResponse response = InfinispanResponse.inReplyTo(request);
+         response.contentAsBytes(mediaType.getOutputPrinter().print(cacheName, keys));
```

Comment:
I wonder if we could send multiple responses for this? There could be a large number of keys and it would be nice to send some to not also consume a large amount of memory on the server at once too.